src/notes.py

```diff
@@ -52,35 +52,55 @@ def add_note(self, note: Note):
         title = note.title.value
         if title in self.data:
             raise NoteExistsError()
-        self.data[title] = note
+        self.data[title.lower()] = note
 
     def edit_note(self, title, new_title: str = '', new_desc: str = '', new_tags: list = []):
-        current_note = self.data[title]
+        current_note = self.data[title.lower()]
 
         if not current_note:
             raise NoteNotExistError()
 
+        old_title = current_note.title.value.lower()
+
         if new_title:
             current_note.title = Title(new_title)
         if new_desc:
             current_note.desc = Desc(new_desc)
         if new_tags:
             current_note.tags = [Tag(tag) for tag in new_tags]
 
-        self.data[current_note.get('title').value] = current_note
-        return
+        if new_title and old_title != new_title.lower():
+            del self.data[old_title]
+            self.data[new_title.lower()] = current_note
+        else:
+            self.data[old_title] = current_note
 
     def find_notes(self, key_words: list) -> list:
-        title = ' '.join(str(word) for word in key_words).lower()
+        notes_by_title = self.__find_notes_by_title(key_words)
+        notes_by_tags = self.__find_notes_by_tags(key_words)
 
+        return notes_by_title | notes_by_tags
+
+    def __find_notes_by_title(self, key_words: list):
+        title = ' '.join(str(word) for word in key_words).lower()
         result = dict()
```

Comment:
why do we need dictionary here?

Reply:
We have a dict format for list of notes, so I think using it in the same format when we use search is a good idea